### ISSUE TITLE
validation: request and response

### ISSUE THREAD

**Issue by mbana** [go-swagger/go-swagger#1738]
## Problem statement

Given I have some JSON for the request and respone I'd like to validate this against a Swagger file. I am currently using a tool written in Node.js soley because there exists this tool, https://github.com/apigee-127/swagger-tools, that allows me to do this.

Effectively, there is a function in the api, in newer versions of the code case, or a middleware, in older versions of the code base that does validation only. One simply gives it the request and response and the library returns a list of validation errors, if any.

Basically, I was wrondering how one would do:

```go
request := <some_json>
response := <some_json>
schema := <some_schema>
validate.DoValidate(schema, request, response)
...
```

**Comment by fredbi** [comment]
You may use the go-openapi/validate package to validate JSON data against schema.

See example here: https://godoc.org/github.com/go-openapi/validate#example-AgainstSchema

**Comment by mbana** [comment]
@fredbi Thanks for the reply.

Would you mind adding more examples. Imagine my request and response look like this, I want to validate based on the path - the libraries I linked to above, have a concept of operation which you retrieve using the path. E.g.,

```json
    {
        "request": {
            "method": "POST",
            "url": "https://modelobank2018.o3bank.co.uk:4501/open-banking/v1.1/payments",
            "path": "/open-banking/v1.1/payments",
            "headers": {
                "user-agent": "node-superagent/3.8.3",
                "authorization": "Bearer 597759d2-790d-4ac2-b0a7-603ff26e5158",
                "content-type": "application/json; charset=utf-8",
                "accept": "application/json; charset=utf-8",
                "x-fapi-interaction-id": "48d657df-12ac-4035-b359-f653a551280c",
                "x-fapi-financial-id": "0015800001041RHAAY",
                "x-idempotency-key": "a8b43b3d-b797-43fe-aea8-3651da3ebb46",
                "x-validation-run-id": "df4ab220-b5cc-11e8-98fd-8c8590302423"
            },
            "body": {
                "Data": {
                    "Initiation": {
                        "InstructionIdentification": "253fcfb5-ae0f-458c-a65b-ef0d3a7a82",
                        "EndToEndIdentification": "aa71fc90-ec55-4ba0-958f-dd274133d2",
                        "InstructedAmount": {
                            "Currency": "GBP",
                            "Amount": "10.00"
                        },
                        "CreditorAccount": {
                            "SchemeName": "SortCodeAccountNumber",
                            "Name": "Sam Morse",
                            "Identification": "11111112345678"
                        }
                    }
                },
                "Risk": {}
            }
        },
        "response": {
            "statusCode": 201,
            "headers": {
                "server": "nginx/1.12.1",
                "date": "Tue, 11 Sep 2018 14:13:36 GMT",
                "content-type": "application/json; charset=utf-8",
                "content-length": "565",
                "connection": "close",
                "x-powered-by": "Express",
                "x-fapi-interaction-id": "48d657df-12ac-4035-b359-f653a551280c",
                "etag": "W/\"235-ayM1zpSIpNY+K2E9KMKVbMl6fcc\""
            },
            "body": {
                "Data": {
                    "Initiation": {
                        "InstructionIdentification": "253fcfb5-ae0f-458c-a65b-ef0d3a7a82",
                        "EndToEndIdentification": "aa71fc90-ec55-4ba0-958f-dd274133d2",
                        "InstructedAmount": {
                            "Currency": "GBP",
                            "Amount": "10.00"
                        },
                        "CreditorAccount": {
                            "SchemeName": "SortCodeAccountNumber",
                            "Name": "Sam Morse",
                            "Identification": "11111112345678"
                        }
                    },
                    "Status": "AcceptedTechnicalValidation",
                    "CreationDateTime": "2018-09-11T14:13:36.292Z",
                    "PaymentId": "pmt-346a5828-cf92-40ac-a51a-8df51f1ff7a2"
                },
                "Risk": {},
                "Links": {
                    "Self": "/open-banking/v1.1/payments/pmt-346a5828-cf92-40ac-a51a-8df51f1ff7a2"
                },
                "Meta": {}
            }
        }
    }
```

When asked to validate the request, the validate method in the library finds the part of the swagger to validate against the request. The only thing I need to specify is the path, which in this case is `/payments`. then the library detects the right portion of the swagger file to validation, which is https://github.com/OpenBankingUK/payment-initiation-api-spec/blob/master/dist/v1.1/payment-initiation-swagger.json#L25.

Hope that made sense. If not, I can expand further.
